src/components/radio/radio.ts

```diff
@@ -0,0 +1,140 @@
+import { html, LitElement } from 'lit';
+import { property, query } from 'lit/decorators.js';
+import { styles } from './radio.material.css';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import { Constructor } from '../common/mixins/constructor.js';
+import { watch } from '../common/decorators';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { live } from 'lit/directives/live.js';
+
+let nextId = 0;
+
+export interface IgcRadioEventMap {
+  igcChange: CustomEvent<void>;
+  igcFocus: CustomEvent<void>;
+  igcBlur: CustomEvent<void>;
+}
+
+export class IgcRadioComponent extends EventEmitterMixin<
+  IgcRadioEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  static styles = styles;
+
+  private inputId = `radio-${nextId++}`;
+  private labelId = `radio-label-${this.inputId}`;
+
+  @query('input[type="radio"]')
```

Comment:
This could be cached.
```suggestion
  @query('input[type="radio"]', true)
```